Content.Client/Overlays/ShowHealthIconsSystem.cs

```diff
@@ -0,0 +1,68 @@
+using Content.Shared.Damage;
+using Content.Shared.Overlays;
+using Content.Shared.StatusIcon;
+using Content.Shared.StatusIcon.Components;
+using Robust.Shared.Prototypes;
+
+namespace Content.Client.Overlays
+{
+    public sealed class ShowHealthIconsSystem : ComponentAddedOverlaySystemBase<ShowHealthIconsComponent>
+    {
+        [Dependency] private readonly IPrototypeManager _prototypeMan = default!;
+        [Dependency] private readonly IEntityManager _entManager = default!;
+
+        public List<string> DamageContainers = new();
+
+        private StatusIconPrototype? _healthyIcon;
```

Comment:
Don't cache the prototype. In general prototypes shouldn't be cached unless they really need to be, and in that case you need to handle prototype realoading.